UnicornEngine/video/source/vulkan/ShaderProgram.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/vulkan/ShaderProgram.hpp>
+#include <unicorn/utility/asset/SimpleStorage.hpp>
+#include <unicorn/utility/Logger.hpp>
+#include <unicorn/video/geometry/Mesh.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+namespace vulkan
+{
+ShaderProgram::ShaderProgram(vk::Device device, const std::string& vertShader, const std::string& fragShader)
+    : m_isCreated(false), m_device(device)
+{
+    unicorn::utility::asset::SimpleStorage& storage = unicorn::utility::asset::SimpleStorage::Instance();
+    unicorn::utility::asset::Handler simpleVertShaderHandler = storage.Get("data/shaders/shader.vert.spv");
+    unicorn::utility::asset::Handler simpleFragShaderHandler = storage.Get("data/shaders/shader.frag.spv");
+
+    if (!simpleVertShaderHandler.IsValid() || !simpleFragShaderHandler.IsValid())
+    {
+        LOG_ERROR("Can't find shaders!");
+        m_isCreated = false;
+    }
+
+    bool shadersCreatedFailed = !CreateShaderModule(simpleVertShaderHandler.GetContent().GetBuffer(), m_vertShaderModule) ||
+                                !CreateShaderModule(simpleFragShaderHandler.GetContent().GetBuffer(), m_fragShaderModule);
+
+    if (shadersCreatedFailed)
+    {
+        LOG_ERROR("Can't create shader module!");
+        m_isCreated = false;
+    }
+
+    vk::PipelineShaderStageCreateInfo vertShaderStageInfo = {};
+    vertShaderStageInfo.stage = vk::ShaderStageFlagBits::eVertex;
+    vertShaderStageInfo.module = m_vertShaderModule;
+    vertShaderStageInfo.pName = "main";
```

Comment:
Yes, but I think will implement this in "Technique feature task" when we will add more then one shader in program